modules/thermo-cycler/thermo-cycler-arduino/thermo-cycler-arduino.ino

```diff
@@ -625,6 +625,58 @@ void read_from_serial() {
     }
 }
 #endif
+
+/* Set up a timer interrupt to check lid switches' status every 20ms.
+ * Keeping the frequency high helps in getting rid of switch debounce. The timer ISR
+ * sets a flag when it's time to check the status of the switches. The loop()
+ * checks this flag and calls lid.switch_poller
+ */
+void setup_timer_interrupt()
+{
+  /* 48MHz/240 = 200kHz (8-bit divisor) | Select Generic Clock (GCLK) 4*/
+  REG_GCLK_GENDIV = GCLK_GENDIV_DIV(240) | GCLK_GENDIV_ID(4);
```

Comment:
Where's the boardfile that provides these included? Should probably explicitly include it here.

[request, not required] Also, could we consider using [CMSIS](https://developer.arm.com/embedded/cmsis) for this? It's the standard cortex HAL and is a little more structured than your standard register bit twiddling. It looks like it's in the arduino library somewhere per a quick `find`.

Reply:
arduino library has it for external hardware interrupts. Couldn't find anything for configuring timer interrupts